### ISSUE TITLE
lint/a11y: Prefer native element rule incorrectly suggests <output> instead of role="status" for loading indicators

### ISSUE THREAD

**Issue by PaalTB** [biomejs/biome#9245]
### Environment information

<details>

```bash
  × The elements with this role can be changed to the following elements:
    <output>
  
    23 │         aria-live="polite"
    24 │         className="container mx-auto max-w-3xl px-4 py-6"
  > 25 │         role="status"
       │         ^^^^^^^^^^^^^
    26 │       >
    27 │         <span className="sr-only">Loading dashboard</span>
  
  i For examples and more information, see WAI-ARIA Roles
  

Checked 1 file in 6ms. No fixes applied.
Found 1 error.
check 

```

</details>

### Environment
@biomejs/biome: 2.4.4
React + TSX

### What happened?

`lint/a11y/useSemanticElements` currently suggests using `<output>` instead of a `<div role="status">` for status messages such as loading indicators.

This conflicts with ARIA guidance to:
- Use native HTML elements when they match the needed semantics, but reserve `<output>` for calculation results / form-related output, not generic status messages.
- A generic “Loading dashboard…” indicator is a status message and is not the result of a calculation or form interaction. 
- ARIA’s own guidance is “prefer native semantics when they fit; otherwise use roles”, not “replace all role="status" with `<output>`”.

In other words, `<output>` is not the correct replacement for every role="status".

Here is a similar discussion on [sonarsource](https://community.sonarsource.com/t/typescript-rspec-6819-should-not-always-require-htmls-output-element-instead-of-aria-status-role/133611l)

### Minimal repro
```tsx
<div role="status" aria-busy="true" aria-live="polite">
  <span className="sr-only">Loading dashboard</span>
  <Skeleton className="h-8 w-52 rounded-md" />
</div>
```

### Expected result

Should not suggest `<output>` for a generic status message like a loading indicator.

The rule should either:
- Only suggest `<output>` when the element clearly represents the result of a calculation or form output, or
- Provide a configuration option to disable this specific suggestion.

### Suggested change
Adjust the rule (likely useSemanticElements or related a11y rule) so that:
- It does not recommend <output> as a blanket replacement for role="status".
- It considers context (e.g., form association, for/name attributes, calculator-like UI) before recommending `<output>`.

Alternatively, downgrade this to a weaker recommendation or allow opting out via rule configuration.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ruidosujeira** [comment]
Good catch! The current role="status" → <output> mapping is indeed incorrect for most use cases. The <output> element is semantically meant for calculation results and form interactions, not for generic status messages like loading indicators.

I'll work on a PR to fix this. The approach I'm considering is removing the role="status" → <output> mapping from the rule entirely, since <div role="status"> is the correct pattern for the vast majority of scenarios and suggesting <output> as a blanket replacement does more harm than good.

Anyway, I'll check it out more carefully.

**Comment by ruidosujeira** [comment]
The `useSemanticElements` rule treats `baseConcepts` (direct equivalents) and `relatedConcepts` (only related ones) as if they were the same thing, suggesting both as valid replacements. The "status" role does not have any HTML `baseConcept`. 

the `<output>` appears only as a `relatedConcept`, because semantically it represents the result of calculations, not generic statuses. The correction would be to add "status" to the exception list (as is already done for "img", "combobox", "listbox", and "option") or rethink how `relatedConcepts` are handled by the rule.

**Comment by dyc3** [comment]
It feels like we probably shouldn't be considering `relatedConcepts` for this rule. Feel free to assess that @ruidosujeira, would love to hear your thoughts.

**Comment by ruidosujeira** [comment]
@dyc3 Agreed. relatedConcepts shouldn't be considered as a valid replacement suggestion, they're only related, not direct equivalents. The cleanest fix would be to change the rule to only use baseConcepts when suggesting elements. This resolves this case and prevents similar ones.

**Comment by dyc3** [comment]
Alright lets do that then. Feel free to send a PR for it.
